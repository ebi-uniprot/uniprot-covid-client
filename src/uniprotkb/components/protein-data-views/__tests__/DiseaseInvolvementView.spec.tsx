import React from 'react';
import { render } from '@testing-library/react';
import DiseaseInvolvement from '../DiseaseInvolvementView';
import DiseaseInvolvementUIDataJson from './__mocks__/diseaseInvolvementUIData.json';

describe('DiseaseInvolvement', () => {
  test('should render DiseaseInvolvement', () => {
    const { asFragment } = render(
      <DiseaseInvolvement
        comments={DiseaseInvolvementUIDataJson}
        primaryAccession="P05067"
        includeTitle
      />
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
