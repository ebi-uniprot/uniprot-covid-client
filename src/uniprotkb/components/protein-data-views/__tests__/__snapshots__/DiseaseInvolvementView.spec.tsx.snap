// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DiseaseInvolvement should render DiseaseInvolvement 1`] = `
<DocumentFragment>
  <h3>
    Involvement in disease
  </h3>
  <h3>
    Alzheimers (AD)
  </h3>
  <span
    class="text-block"
  />
  <div
    class="info-list"
  >
    <div
      class="info-list__item"
    >
      <div
        class="info-list__title"
      >
        <h5
          class="bold"
        >
          Note
        </h5>
      </div>
      <div
        class="info-list__content"
      >
        <ul
          class="expandable-list no-bullet"
        >
          <li>
            Some disease note
          </li>
        </ul>
      </div>
    </div>
    <div
      class="info-list__item"
    >
      <div
        class="info-list__title"
      >
        <h5
          class="bold"
        >
          Description
        </h5>
      </div>
      <div
        class="info-list__content"
      >
        Some description
      </div>
    </div>
    <div
      class="info-list__item"
    >
      <div
        class="info-list__title"
      >
        <h5
          class="bold"
        >
          See also
        </h5>
      </div>
      <div
        class="info-list__content"
      >
        (
        <a
          class="external-link"
          href="https://www.ebi.ac.uk/ena/data/view/some ref id"
          rel="noopener"
          target="_blank"
        >
          EMBL
          <svg
            fill="none"
            viewBox="0 0 16 16"
            width="12.5"
          >
            <path
              d="M16 0h-6v2h3.071L7 8.071l1.414 1.414L14 3.9V6h2V0z"
              fill="currentColor"
            />
            <path
              d="M7 2H0v14h14V9h-2v5H2V4h5V2z"
              fill="currentColor"
            />
          </svg>
        </a>
         | 
        <a
          class="external-link"
          href="https://www.ncbi.nlm.nih.gov/protein/some ref id"
          rel="noopener"
          target="_blank"
        >
          GenBank
          <svg
            fill="none"
            viewBox="0 0 16 16"
            width="12.5"
          >
            <path
              d="M16 0h-6v2h3.071L7 8.071l1.414 1.414L14 3.9V6h2V0z"
              fill="currentColor"
            />
            <path
              d="M7 2H0v14h14V9h-2v5H2V4h5V2z"
              fill="currentColor"
            />
          </svg>
        </a>
         | 
        <a
          class="external-link"
          href="http://getentry.ddbj.nig.ac.jp/getentry/dad/some ref id"
          rel="noopener"
          target="_blank"
        >
          DDBJ
          <svg
            fill="none"
            viewBox="0 0 16 16"
            width="12.5"
          >
            <path
              d="M16 0h-6v2h3.071L7 8.071l1.414 1.414L14 3.9V6h2V0z"
              fill="currentColor"
            />
            <path
              d="M7 2H0v14h14V9h-2v5H2V4h5V2z"
              fill="currentColor"
            />
          </svg>
        </a>
        ) 
        <a
          class="external-link"
          href="//www.ebi.ac.uk/ena/data/view/some ref id"
          rel="noopener"
          target="_blank"
        >
          some ref id
          <svg
            fill="none"
            viewBox="0 0 16 16"
            width="12.5"
          >
            <path
              d="M16 0h-6v2h3.071L7 8.071l1.414 1.414L14 3.9V6h2V0z"
              fill="currentColor"
            />
            <path
              d="M7 2H0v14h14V9h-2v5H2V4h5V2z"
              fill="currentColor"
            />
          </svg>
        </a>
        [
        <a
          href="#some isoform"
        >
          some isoform
        </a>
        ]
      </div>
    </div>
  </div>
</DocumentFragment>
`;
